pkg/config/types_test.go

```diff
@@ -0,0 +1,181 @@
+package config
+
+import (
+	"reflect"
+	"testing"
+)
+
+func Test_parseRbac(t *testing.T) {
+	type args struct {
+		in string
+	}
+	tests := []struct {
+		name string
+		args args
+		want []string
+	}{{
+		args: args{""},
+		want: nil,
+	}, {
+		args: args{"abc"},
+		want: []string{"abc"},
+	}, {
+		args: args{" abc "},
+		want: []string{"abc"},
+	}, {
+		args: args{"abc,def"},
+		want: []string{"abc", "def"},
+	}, {
+		args: args{"abc,,,def,"},
+		want: []string{"abc", "def"},
+	}, {
+		args: args{"abc, def"},
+		want: []string{"abc", "def"},
+	}, {
+		args: args{"abc ,def "},
+		want: []string{"abc", "def"},
+	}}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := parseRbac(tt.args.in); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("parseRbac() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_parseKinds(t *testing.T) {
+	type args struct {
+		in string
+	}
+	tests := []struct {
+		name string
+		args args
+		want []filter
+	}{{
+		args: args{""},
+		want: []filter{},
+	}, {
+		args: args{"[]"},
+		// TODO: this looks strange
+		want: []filter{
+			{},
+		},
+	}, {
+		args: args{"[*]"},
+		want: []filter{
+			{"*", "", ""},
+		},
+	}, {
+		args: args{"[Node]"},
+		want: []filter{
+			{"Node", "", ""},
+		},
+	}, {
+		args: args{"[Node,*,*]"},
+		want: []filter{
+			{"Node", "*", "*"},
+		},
+	}, {
+		args: args{"[Pod,default,nginx]"},
+		want: []filter{
+			{"Pod", "default", "nginx"},
+		},
+	}, {
+		args: args{"[Pod,*,nginx]"},
+		want: []filter{
+			{"Pod", "*", "nginx"},
+		},
+	}, {
+		args: args{"[Pod,*]"},
+		want: []filter{
+			{"Pod", "*", ""},
+		},
+	}, {
+		args: args{"[Pod,default,nginx][Pod,kube-system,api-server]"},
+		want: []filter{
+			{"Pod", "default", "nginx"},
+			{"Pod", "kube-system", "api-server"},
+		},
+	}, {
+		args: args{"[Pod,default,nginx],[Pod,kube-system,api-server]"},
+		want: []filter{
+			{"Pod", "default", "nginx"},
+			{"Pod", "kube-system", "api-server"},
+		},
+	}, {
+		args: args{"[Pod,default,nginx] [Pod,kube-system,api-server]"},
+		want: []filter{
+			{"Pod", "default", "nginx"},
+			{"Pod", "kube-system", "api-server"},
+		},
+	}, {
+		args: args{"[Pod,default,nginx]Pod,kube-system,api-server[Pod,kube-system,api-server]"},
+		want: []filter{
+			{"Pod", "default", "nginx"},
+			{"Pod", "kube-system", "api-server"},
+		},
```

Comment:
Looks strange that we can add anything between `][`.

Reply:
Can we deny this?